### ISSUE TITLE
[JENKINS-16521] Multi-Project Throttle Categories don't work with Matrix jobs

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/matrix-project-plugin#532]
<p>I've set up two Multi-Project Throttle Categories, filesystem and postgres, both with maximum total concurrent builds of 0 and maximum concurrent builds per node of 1.</p>

<p>This is because only one job that uses these categories can run on a host at once, if two try, they'll stomp all over each other.</p>

<p>So, I've configured two jobs, both matrix jobs, and both configured with "Throttle Concurrent Builds" checked, "throttle this projects as part of one or more catogories" checked and both the categories defined above checked.</p>

<p>However, when a node has more than one executor, I'll frequently see multiple configurations from the matrix being run on the same node at the same time. Sure enough, they stomp all over each other. </p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cjw296">cjw296</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521" target="_blank">Multi-Project Throttle Categories don't work with Matrix jobs</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=oleg_nenashev">oleg_nenashev</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: matrix-project-plugin, throttle-concurrent-builds-plugin
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2014-06-01T13:15:39+00:00
<li><b>votes</b>: 12
<li><b>watchers</b>: 20
<li><b>imported</b>: 2025-12-06
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>I've set up two Multi-Project Throttle Categories, filesystem and postgres, both with maximum total concurrent builds of 0 and maximum concurrent builds per node of 1.</p>

<p>This is because only one job that uses these categories can run on a host at once, if two try, they'll stomp all over each other.</p>

<p>So, I've configured two jobs, both matrix jobs, and both configured with "Throttle Concurrent Builds" checked, "throttle this projects as part of one or more catogories" checked and both the categories defined above checked.</p>

<p>However, when a node has more than one executor, I'll frequently see multiple configurations from the matrix being run on the same node at the same time. Sure enough, they stomp all over each other. </p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
master and 9 slaves all running SUSE Linux.<br/>
Jenkins 1.480.2<br/>
throttle-concurrents: 1.7.2
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=172831&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172831">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=172831&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172831">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This issue would appear to be related:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-12240" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-12240</a></p>

<p>...but suggests a desire for the opposite behaviour: as many configurations of a particular job as possible should run on a node.</p>

<p>However, I'm sure there are just as many people who want the behaviour I'm describing, here's an example:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-13619" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-13619</a></p>
</pre>
</details>

</ul>
</details>

<p>This issue would appear to be related:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-12240" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-12240</a></p>

<p>...but suggests a desire for the opposite behaviour: as many configurations of a particular job as possible should run on a node.</p>

<p>However, I'm sure there are just as many people who want the behaviour I'm describing, here's an example:</p>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-13619" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-13619</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=172834&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172834">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=172834&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-172834">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I've just verified that this is a regression in 1.7.2.</p>

<p>1.7.1 behaves as expected, so I'll downgrade to that as a workaround...</p>
</pre>
</details>

</ul>
</details>

<p>I've just verified that this is a regression in 1.7.2.</p>

<p>1.7.1 behaves as expected, so I'll downgrade to that as a workaround...</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=173071&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173071">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=173071&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-173071">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-16552" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-16552</a> is a duplicate of this issue.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins-ci.org/browse/JENKINS-16552" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-16552</a> is a duplicate of this issue.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>wael:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=175317&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175317">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=175317&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-175317">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I tried 1.7.1 but had the same unwanted behaviour as 1.7.2!</p>
</pre>
</details>

</ul>
</details>

<p>I tried 1.7.1 but had the same unwanted behaviour as 1.7.2!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=177899&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-177899">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=177899&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-177899">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'm also now seeing this on 1.7.1 with 1.480.3 :-/</p>
</pre>
</details>

</ul>
</details>

<p>I'm also now seeing this on 1.7.1 with 1.480.3 :-/</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>judoole:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=178705&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178705">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=178705&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178705">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Same here. Version 1.72 and 1.513 of Jenkins. Is this, <a href="https://issues.jenkins-ci.org/browse/JENKINS-12240" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-12240</a>, supposed to fix it? It is included in version 1.72 according to commits. </p>
</pre>
</details>

</ul>
</details>

<p>Same here. Version 1.72 and 1.513 of Jenkins. Is this, <a href="https://issues.jenkins-ci.org/browse/JENKINS-12240" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-12240</a>, supposed to fix it? It is included in version 1.72 according to commits. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>judoole:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=178710&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178710">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=178710&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178710">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Update: Cloned tag 1.7.1 and installed and it seems to work using Category.
Created a "Matrix-job" category using "Maximum Total Concurrent Build = 4" and "Maximum Concurrent Builds Per Node = 4". The job itself is configured with "Maximum Concurrent Builds Per Node = 1" and "Maximum Total Concurrent Build = 4". I am not using any slaves though.</p>
</pre>
</details>

</ul>
</details>

<p>Update: Cloned tag 1.7.1 and installed and it seems to work using Category.<br/>
Created a "Matrix-job" category using "Maximum Total Concurrent Build = 4" and "Maximum Concurrent Builds Per Node = 4". The job itself is configured with "Maximum Concurrent Builds Per Node = 1" and "Maximum Total Concurrent Build = 4". I am not using any slaves though.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=178815&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178815">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=178815&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-178815">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yeah, I thought it was working, and then found overlapping matrix child jobs on some slave nodes.
So, have to default to one executor per slave, which means those slaves can't be used for any other jobs (that don't use the throttled resources), which is a waste <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Yeah, I thought it was working, and then found overlapping matrix child jobs on some slave nodes.<br/>
So, have to default to one executor per slave, which means those slaves can't be used for any other jobs (that don't use the throttled resources), which is a waste <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/sad.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hiteswar_kumar:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=179145&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179145">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=179145&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-179145">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi Andrew(Abayer),
Can you please fix this and release 1.7.3 fast.
or if any one has already done fix..please do the pull request to get it merged.</p>

</pre>
</details>

</ul>
</details>

<p>Hi Andrew(Abayer),<br/>
Can you please fix this and release 1.7.3 fast.<br/>
or if any one has already done fix..please do the pull request to get it merged.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=188337&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-188337">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=188337&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-188337">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Reassigned the issue</p>
</pre>
</details>

</ul>
</details>

<p>Reassigned the issue</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>swiksted:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=194730&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-194730">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=194730&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-194730">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,
is this behaviour somehow related to issue we've see lately with matrix jobs and throttle-plugin: throttle plugin do not recognize slave nodes that has been  added <em>after</em> the matrix jobs has been gone to queue. Resulting those jobs going to slave nodes that has been there before job launch and new nodes doing nothing.</p>
</pre>
</details>

</ul>
</details>

<p>Hi,<br/>
is this behaviour somehow related to issue we've see lately with matrix jobs and throttle-plugin: throttle plugin do not recognize slave nodes that has been  added <em>after</em> the matrix jobs has been gone to queue. Resulting those jobs going to slave nodes that has been there before job launch and new nodes doing nothing.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>cjw296:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=195009&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-195009">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=195009&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-195009">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>No, that sounds entirely unrelated. Please file a separate issue for that.</p>
</pre>
</details>

</ul>
</details>

<p>No, that sounds entirely unrelated. Please file a separate issue for that.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=196340&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-196340">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=196340&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-196340">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'd say that the issue completely duplicates <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a>.</p>

<p>BTW, I've started working on the configurable behavior for Matrix projects. It should resolve both issues.</p>
</pre>
</details>

</ul>
</details>

<p>I'd say that the issue completely duplicates <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a>.</p>

<p>BTW, I've started working on the configurable behavior for Matrix projects. It should resolve both issues.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>andrea_curtoni:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=200417&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200417">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=200417&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-200417">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Hi,
 we have another issue with matrix and throttle-concurrents, I cannot say if it is related to this.</p>

<p>We have 4 executors on a node and 2 matrix projects using them. Each matrix project spawns 4 jobs. Jobs from the same matrix project can run in parallel on the same node but they conflict with jobs from other projects. So we configured the two jobs with the same category and specified to run 1 job per node. Unfortunately jobs from different projects in the same category are spawned to the same node.
We would expect to have 4 jobs from the same project... or at least 1 job from a single project and 3 free executors, not mixed project jobs.</p>

<p>We did the following:</p>

<p>Global configuration:
 Category Name: mock</p>
<ul class="alternate" type="square">
	<li>Maximum Total Concurrent Builds: 0</li>
	<li>Maximum Concurrent Builds Per Node: 1</li>
</ul>


<p>Job configurations:
 Throttle Concurrent Builds =&gt; checked
 Throttle this project as part of one or more categories =&gt; selected</p>
<ul class="alternate" type="square">
	<li>Maximum Total Concurrent Builds: 0</li>
	<li>Maximum Concurrent Builds Per Node: 1</li>
	<li>Multi-Project Throttle Category: mock</li>
</ul>


<p>We also tried to specify the per-node global conf without any help. We also tried to set the limit for the master node (which is used as the main executor for the matrix jobs). No luck.</p>

<p>Note the same configuration with other non-matrix projects works fine.</p>

<p>Jenkins version: 1.559
Throttle Concurrent Builds Plug-in version: 1.8.2</p>

<p>Should I open another issue for this problem?</p>
</pre>
</details>

</ul>
</details>

<p>Hi,<br/>
 we have another issue with matrix and throttle-concurrents, I cannot say if it is related to this.</p>

<p>We have 4 executors on a node and 2 matrix projects using them. Each matrix project spawns 4 jobs. Jobs from the same matrix project can run in parallel on the same node but they conflict with jobs from other projects. So we configured the two jobs with the same category and specified to run 1 job per node. Unfortunately jobs from different projects in the same category are spawned to the same node.<br/>
We would expect to have 4 jobs from the same project... or at least 1 job from a single project and 3 free executors, not mixed project jobs.</p>

<p>We did the following:</p>

<p>Global configuration:<br/>
 Category Name: mock</p>
<ul class="alternate" type="square">
	<li>Maximum Total Concurrent Builds: 0</li>
	<li>Maximum Concurrent Builds Per Node: 1</li>
</ul>


<p>Job configurations:<br/>
 Throttle Concurrent Builds => checked<br/>
 Throttle this project as part of one or more categories => selected</p>
<ul class="alternate" type="square">
	<li>Maximum Total Concurrent Builds: 0</li>
	<li>Maximum Concurrent Builds Per Node: 1</li>
	<li>Multi-Project Throttle Category: mock</li>
</ul>


<p>We also tried to specify the per-node global conf without any help. We also tried to set the limit for the master node (which is used as the main executor for the matrix jobs). No luck.</p>

<p>Note the same configuration with other non-matrix projects works fine.</p>

<p>Jenkins version: 1.559<br/>
Throttle Concurrent Builds Plug-in version: 1.8.2</p>

<p>Should I open another issue for this problem?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=202818&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202818">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=202818&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202818">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>@Andrea
This is a similar issue.
I'll provide a Dev Build of the plugin in <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a></p>
</pre>
</details>

</ul>
</details>

<p>@​Andrea<br/>
This is a similar issue.<br/>
I'll provide a Dev Build of the plugin in <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/oleg_nenashev.png" title="oleg_nenashev's avatar" /> oleg_nenashev:</i></summary>

<ul>
<li><i><a class="no-jira-link-rewrite" href="https://issue-redirect.jenkins.io/issue/16521?focusedId=202819&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202819">Comment link</a></i>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-16521?focusedId=202819&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-202819">comment link</a> (no redirect)</i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Marking as duplicate.
Let's track the further activity in <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a> </p>
</pre>
</details>

</ul>
</details>

<p>Marking as duplicate.<br/>
Let's track the further activity in <a href="https://issue-redirect.jenkins.io/issue/13619" title="Add ability to throttle number of concurrent builds of matrix configurations per node" class="issue-link" data-issue-key="JENKINS-13619"><del>JENKINS-13619</del></a> </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/matrix-project-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-13619%22">JENKINS-13619</a>]</i>
